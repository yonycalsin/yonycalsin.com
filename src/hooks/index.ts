export { default as useMDXComponent } from './use-mdx-component'
export { default as useQueryCache } from './use-query-cache'
