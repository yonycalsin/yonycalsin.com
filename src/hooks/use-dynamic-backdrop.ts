import * as React from 'react'

function useDynamicBackdrop() {
  const backdropRef = React.useRef<HTMLDivElement>(null)

  const handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const element = event.target as HTMLElement

    if (!(element instanceof HTMLElement) || !backdropRef.current) {
      return
    }

    const { left, top, width, height } = element.getBoundingClientRect()

    backdropRef.current.style.setProperty('--left', `${left}px`)

    backdropRef.current.style.setProperty('--top', `${top}px`)

    backdropRef.current.style.setProperty('--width', `${width}px`)

    backdropRef.current.style.setProperty('--height', `${height}px`)

    backdropRef.current.style.opacity = '1'

    backdropRef.current.style.visibility = 'visible'
  }

  const handleMouseLeave = () => {
    if (!backdropRef.current) {
      return
    }

    backdropRef.current.style.opacity = '0'

    backdropRef.current.style.visibility = 'hidden'
  }

  return { backdropRef, handleMouseEnter, handleMouseLeave }
}

export default useDynamicBackdrop
