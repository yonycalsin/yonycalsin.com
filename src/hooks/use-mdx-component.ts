import * as React from 'react'
import _jsx_runtime from 'react/jsx-runtime'
import ReactDOM from 'react-dom'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
function getMdxComponent(code: string): React.ComponentType<any> {
  const scope = {
    React,
    ReactDOM,
    _jsx_runtime,
  }

  const fn = new Function(...Object.keys(scope), code)

  return fn(...Object.values(scope)).default
}

function useMDXComponent(code: string) {
  return React.useMemo(() => getMdxComponent(code), [code])
}

export default useMDXComponent
