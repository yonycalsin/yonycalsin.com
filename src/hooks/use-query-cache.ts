import { QueryKey, useQueryClient } from '@tanstack/react-query'

function useQueryCache<T = unknown>(queryKey: QueryKey) {
  const queryClient = useQueryClient()

  const queryCache = queryClient.getQueryCache()

  const queryCacheValue = queryCache.find<T>(queryKey, {
    exact: true,
  })

  return queryCacheValue
}

export default useQueryCache
