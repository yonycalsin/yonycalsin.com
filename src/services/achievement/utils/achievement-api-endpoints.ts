import env from '~/utils/constants/env'

export const achievementApiEndpoints = {
  ALL_ACHIEVEMENTS: `${env.REST_API_URL}/achievements`,
  FEATURED_ACHIEVEMENTS: `${env.REST_API_URL}/achievements?isFeatured=true`,
}
