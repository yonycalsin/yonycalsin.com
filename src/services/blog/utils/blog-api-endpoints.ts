import env from '~/utils/env'

export const blogApiEndpoints = {
  POST: (postSlug: string) => `${env.REST_API_URL}/blog/posts/${postSlug}`,
  POSTS: `${env.REST_API_URL}/blog/posts`,
  CATEGORY: (categorySlug: string) => `${env.REST_API_URL}/blog/categories/${categorySlug}`,
  CATEGORIES: `${env.REST_API_URL}/blog/categories`,
}
