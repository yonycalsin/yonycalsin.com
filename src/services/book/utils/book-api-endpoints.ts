import env from '~/utils/constants/env'

export const bookApiEndpoints = {
  ALL_BOOKS: `${env.REST_API_URL}/books`,
  READING_BOOKS: `${env.REST_API_URL}/books?status=Reading`,
}
