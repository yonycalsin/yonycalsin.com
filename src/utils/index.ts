export * from './mdx'
export * from './toggled'

// @yonycalsin/eslint-import-sort/typescript-react-exports group
export { default as getRandomBadgeColors } from './get-random-badge-colors'
export { default as normalizeDisplayUrl } from './normalize-display-url'
