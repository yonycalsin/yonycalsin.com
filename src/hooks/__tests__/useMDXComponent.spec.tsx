import { render, renderHook, screen } from '@testing-library/react'

import { useMDXComponent } from 'hooks'
import { buildMdxRuntimeCode } from 'utils'

describe('useMDXComponent', () => {
  it('returns the mdx processed code', () => {
    const utils = renderHook(() => useMDXComponent(buildMdxRuntimeCode(`jsx('p', {children: 'Hello World'})`)))

    const Component = utils.result.current

    expect(Component).toStrictEqual(expect.any(Function))

    const view = render(<Component />)

    const content = screen.queryByText(/hello world/i)

    expect(content).toBeInTheDocument()

    expect(view.baseElement).toMatchSnapshot()
  })
})
