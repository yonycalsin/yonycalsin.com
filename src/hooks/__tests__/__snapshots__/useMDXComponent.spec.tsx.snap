// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hooks / useMDXComponent returns the mdx processed code 1`] = `
<body>
  <div />
  <div>
    <p>
      Hello World
    </p>
  </div>
</body>
`;
